src/utils/xmlfile.cpp

```diff
@@ -532,84 +532,61 @@ template<typename T> bool XMLfile::Node::getValue(T& value) const
 {
 	if(m_xmlnode)
 	{
-		//value=T(m_xmlnode->value());
 		std::stringstream ss(m_xmlnode->value());
-		ss>>value;
+		// Check if input has correct sign
+		if (std::is_unsigned_v<T>) {
+			if (ss.str().find_first_of("-") != std::string::npos) {
+				std::cerr << "ERROR parsing \"" << ss.str() << "\" to data type " << typeid(T).name() << " from tag \"<" << name() << ">\" in xml file" << std::endl;
+				std::cerr << "The tag contains a negative value but an unsigned value was expected." << std::endl;
+				Simulation::exit(1);
+			}
+		}
+		ss >> value;
```

Comment:
I am just curious and want to learn something: Why did you put this line (`ss >> value;`) after the first check? In my version, it was directly after the stringstream init.

Reply:
If the type is unsigned and a signed value is in the XML input parsing before is questionable.

Reply:
Ok, I see. Thank you for the clarification :)